grafana/grafana
# Public Dashboards Bug: orgId missing when fetchng pubdash panel data unauthed

**Problem**
We ran into issues when an unauthed user accessed the pubdash panel query endpoint. A user with an OrgId is expected to be passed pretty far down the chain, and was causing failures in a couple places.

**Solution**
Create a blank user model with the OrgId from the public dashboard and pass it down. Also added an integration test because our existing tests wouldn't catch this case due to mocking.

**How to Test**
View a public dashboard in an incognito window. The api call to get panel query data should succeed.

**Notes for Reviewer**
This was my first time writing an integration test. It seemed longer than it had to be for inserting the models I needed into the database. Do we have any factories for creating models for tests quickly? Is there any test database cleanup we need to do? Or does it get cleaned up automatically when the test is done?
